fitbenchmarking/fitbenchmarking
# Added python and numpy version tags to checkpoint.json

## Description of work

Fixes #1318 

Added `python` and `numpy` tags to the fitbenchmarking report. Also updated the style of the report. In this PR, I am not trying to handle for the case when checkpoint files (generated on different numpy versions) are merged. This needs a little while to understand the issue and properly handle that case. In my experience, there are very few cases that will cause an issue. 





## Testing Instructions



1. Run fitbenchmarking a verify that python and numpy versions are visible.

<img width="541" alt="image" src="https://github.com/user-attachments/assets/aacc7b83-2de1-4819-b112-0773504d1d8e">

2. Also verify that the **checkpoint.json** in `fitbenchmarking_results` contains the config.
<img width="328" alt="image" src="https://github.com/user-attachments/assets/0fbae5e3-fac9-49c0-87dd-24b945d3e7f3">

3. Delete the config from the **checkpoint.json** in `fitbenchmarking_results`. Run `fitbenchmarking-cp report`. Verify info_unavaliable is shown in the report.
<img width="680" alt="image" src="https://github.com/user-attachments/assets/b891c309-fb21-4c20-a879-9ac53628515f">

4. Run `pytest fitbenchmarking/utils/tests/test_checkpoint.py -k CheckpointTests`. Verify tests pass.

4. I generated some results in python `3.10`, `3.13` and `3.8`.

<img width="525" alt="image" src="https://github.com/user-attachments/assets/ecde8b05-b306-4a08-956e-34029263851e">
<img width="525" alt="image" src="https://github.com/user-attachments/assets/3f8d6fd6-1ae5-4fa5-ac22-cf96e6d5f5d7">
<img width="525" alt="image" src="https://github.com/user-attachments/assets/e4b2bb67-ba9c-4ef3-a67c-22cda9a6a9c7">

I was able to open most of the checkpoint files in other environments without issues (in one case there was a warning). However, I was not able to open a checkpoint file generated in python `3.13` in python `3.8`. I got this error.

<img width="1081" alt="image" src="https://github.com/user-attachments/assets/0b610389-7cbd-43fa-a3e6-d11ecca29044">




## Checklist
